integration/e2emimir/client.go

```diff
@@ -288,6 +290,29 @@ func (c *Client) LabelValuesCardinality(labelNames []string, selector string, li
 	return c.httpClient.Do(req.WithContext(ctx))
 }
 
+// GetPrometheusMetadata fetches the metadata from the Prometheus endpoint /api/v1/metadata.
+func (c *Client) GetPrometheusMetadata(metric string, limit int) (*http.Response, error) {
+	body := make(url.Values)
+	if len(metric) > 0 {
+		body.Set("metric", metric)
+	}
+	if limit > 0 {
+		body.Set("limit", strconv.Itoa(limit))
+	}
+
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("http://%s/prometheus/api/v1/metadata", c.querierAddress), strings.NewReader(body.Encode()))
```

Comment:
Don't the query parameters (`body`) need to be included in the URL, not the request body?